internal/serialization/serialization.go

```diff
@@ -136,42 +138,76 @@ func (s *Service) FindSerializerFor(obj interface{}) (pubserialization.Serialize
 	return serializer, nil
 }
 
-func (s *Service) registerDefaultSerializers() error {
-	sers := []struct {
-		s pubserialization.Serializer
-		l string
-		i int32
-	}{
-		{l: "uint8", i: ConstantTypeByte, s: &ByteSerializer{}},
-		{l: "bool", i: ConstantTypeBool, s: &BoolSerializer{}},
-		{l: "uint16", i: ConstantTypeUInteger16, s: &UInteger16Serializer{}},
-		{l: "int16", i: ConstantTypeInteger16, s: &Integer16Serializer{}},
-		{l: "int32", i: ConstantTypeInteger32, s: &Integer32Serializer{}},
-		{l: "int64", i: ConstantTypeInteger64, s: &Integer64Serializer{}},
-		{l: "float32", i: ConstantTypeFloat32, s: &Float32Serializer{}},
-		{l: "float64", i: ConstantTypeFloat64, s: &Float64Serializer{}},
-		{l: "string", i: ConstantTypeString, s: &StringSerializer{}},
-		{l: "nil", i: ConstantTypeNil, s: &NilSerializer{}},
-		{l: "[]uint8", i: ConstantTypeByteArray, s: &ByteArraySerializer{}},
-		{l: "[]bool", i: ConstantTypeBoolArray, s: &BoolArraySerializer{}},
-		{l: "[]uint16", i: ConstantTypeUInteger16Array, s: &UInteger16ArraySerializer{}},
-		{l: "[]int16", i: ConstantTypeInteger16Array, s: &Integer16ArraySerializer{}},
-		{l: "[]int32", i: ConstantTypeInteger32Array, s: &Integer32ArraySerializer{}},
-		{l: "[]int64", i: ConstantTypeInteger64Array, s: &Integer64ArraySerializer{}},
-		{l: "[]float32", i: ConstantTypeFloat32Array, s: &Float32ArraySerializer{}},
-		{l: "[]float64", i: ConstantTypeFloat64Array, s: &Float64ArraySerializer{}},
-		{l: "[]string", i: ConstantTypeStringArray, s: &StringArraySerializer{}},
-		{l: "serialization.JSON", i: JSONSerializationType, s: &JSONValueSerializer{}},
-		{l: "!gob", i: GoGobSerializationType, s: &GobSerializer{}},
-	}
-	for _, ser := range sers {
-		if err := s.registerSerializer(ser.s); err != nil {
-			return err
-		}
-		s.nameToID[ser.l] = ser.i
+func (s *Service) LookUpDefaultSerializer(obj interface{}) pubserialization.Serializer {
+	serializer := s.lookupBuiltinSerializer(obj)
+	if serializer != nil {
+		return serializer
 	}
-	err := s.registerIdentifiedFactories()
-	if err != nil {
+	if _, ok := obj.(pubserialization.IdentifiedDataSerializable); ok {
+		return s.registry[s.nameToID["identified"]]
+	}
+	if _, ok := obj.(pubserialization.Portable); ok {
+		return s.registry[s.nameToID["!portable"]]
+	}
+	return nil
+}
+
+func (s *Service) lookupBuiltinDeserializer(typeID int32) pubserialization.Serializer {
+	switch typeID {
+	case TypeNil:
+		return nilSerializer
+	case TypeBool:
```

Comment:
Can't we put portable and identified serializer(maybe even gob and json) ids here, instead of putting them to s.registry and doing lookup there? 
Then, I believe, the only things we put to registry would be the custom serializer and we can rename it as such

Reply:
Yes, removed portable and identified serializers there: https://github.com/hazelcast/hazelcast-go-client/pull/581/commits/646f1893fc76711d3ffecc2c6d09d0253044d316